### PR TITLE
authorization with roles and mutations exposing

### PR DESCRIPTION
**Pull request by vitaliitomkiv**

### CODE REVIEW

File: src/main/java/com/altigee/needoff/auth/model/UserPrincipal.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package com.altigee.needoff.auth.model;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+public class UserPrincipal implements UserDetails {
```
Comment (nestorsokil) [inline src/main/java/com/altigee/needoff/auth/model/UserPrincipal.java:10]: You can use Lombok's `@Getter`

File: src/main/java/com/altigee/needoff/auth/service/AccountServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  @Autowired private AccountRepo accountRepo;
+
+  @Autowired private RoleRepo roleRepo;
+
+  @Override
+  public Optional<Account> grantAccountAdminRoles(Long accountId) {
+
+    return accountRepo
+        .findById(accountId)
+        .map(
```
Comment (nestorsokil) [inline src/main/java/com/altigee/needoff/auth/service/AccountServiceImpl.java:23]: You can use `Stream::peek`, like so `.findById(accountId).peek(account -> account.setRoles(...))...`

File: src/main/java/com/altigee/needoff/auth/service/AuthenticationFacadeImpl.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+@Component
+public class AuthenticationFacadeImpl  implements AuthenticationFacade{
+
+    @Autowired
+    private AccountRepo accountRepo;
+
+    @Override public Optional<Account> getAuthenticatedAccount() {
+
+       Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```
Comment (nestorsokil) [inline src/main/java/com/altigee/needoff/auth/service/AuthenticationFacadeImpl.java]: can `getAuthentication()` returns `null`?

File: src/main/java/com/altigee/needoff/auth/service/AccountServiceImpl.java (hunk 2 above)
Comment (vitaliitomkiv) [reply src/main/java/com/altigee/needoff/auth/service/AccountServiceImpl.java:23]: its `Optional.map` which doesn't have `peek` function

File: src/main/java/com/altigee/needoff/auth/model/UserPrincipal.java (hunk 1 above)
Comment (vitaliitomkiv) [reply src/main/java/com/altigee/needoff/auth/model/UserPrincipal.java:10]: actually no need to introduce getters sine every method is explicitly defined

File: src/main/java/com/altigee/needoff/auth/service/AuthenticationFacadeImpl.java (hunk 3 above)
Comment (vitaliitomkiv) [reply src/main/java/com/altigee/needoff/auth/service/AuthenticationFacadeImpl.java]: refactored to NPE safe

**Review by nestorsokil — APPROVED** [review]

**Pull request merged**